ERCS/erc-7652.md

```diff
@@ -0,0 +1,160 @@
+---
+eip: 7652
+title: ERC-721 Guarantee Extension
+description: ERC-721 Guarantor role interface extension
+author: Liu.C.Dao (@CDao) <iunknow@163.com>, Sam <1047180870@qq.com>
+discussions-to: https://ethereum-magicians.org/t/erc-7652-eip-721-guarantee-extension/19284
+status: Draft
+type: Standards Track
+category: ERC
+created: 2024-03-10
+requires: 165, 721
+---
+
+## Abstract
+
+This specification defines functions outlining a guarantor role for instance of [EIP-721](./eip-721.md). The guarantee interface implements the user-set valuation and guarantee share for a given NFT (token ID), as well as the guarantee rights enjoyed and obligations assumed during subsequent transactions. An implementation enables the user to read or set the current guarantee value for a given NFT (token ID), and also realizes the distribution of guarantee interest and the performance of guarantee obligations. It sends the standardized events when the status changes. This proposal relies on and extends the existing [EIP-721](./eip-721.md).
+
+## Motivation
+
+NFT (token ID) commonly face the issue of insufficient market liquidity: the main reason being the lack of transparency in NFT pricing, making it difficult for users to cash out after trading and purchasing NFT (token ID).
+
+With the introduction of the guarantor role, different guarantor groups can offer various price guarantees for NFT (token ID), establishing a multi-faceted price evaluation system for NFT (token ID).
```

Comment:
I appreciate the explanation. I still didn't quite get what it means. Can you elaborate what "price guarantees for NFT" means?

Reply:
Thank you for your feedback. I appreciate your attention to detail .

Regarding the issue of "price guarantees for NFT ", I understand that without further context, this term may be abstract. Specifically, this feature aims to provide a mechanism to ensure that NFT buyers are not affected by drastic price fluctuations after purchase, thereby increasing the frequency of NFT transactions. An example is as follows:

In a certain NFT transaction, suppose A spends x tokens to purchase an NFT (tokenId) from B, and C provides y tokens as a guarantee for the NFT(tokenId), receiving a transaction commission of z. 
After purchasing the NFT (tokenId), A has an opportunity to exchange the NFT (tokenId) with C for y tokens (this exchange will be automatically executed as long as A applies, without the need for further authorization from C).

In the transaction between A and B, assume A spends 10 tokens to purchase an NFT (tokenId) from B. In this transaction, there are C1, C2... C9, who are nine different guarantors, and their guarantee values for the NFT (tokenId) are 1 token, 2 tokens... 9 tokens, respectively (Here, C1, C2...C9 fairly distribute transaction commissions, and at the same time, the guarantee chain of C1, C2...C9 also improves the price transparency of NFTs): 

Similarly, after purchasing (tokenId), A has an opportunity to exchange NFTs (tokenId) with C9 for 9 tokens, and C9 has an opportunity to exchange NFTs (tokenId) with C8 for 8 tokens... and so on. C2 has an opportunity to exchange NFTs (tokenId) with C1 for 1 token (this guarantee chain further enhances the liquidity and monetization ability of NFTs).
  
If a guarantor C believes that a certain NFT (tokenId) is worth 10 tokens but only has 5 tokens, then C can organize a DAO that believes the NFT is worth 10 tokens. When the DAO raises 10 tokens, the DAO can participate as a guarantor.
  
If necessary, I would like to add these explanations to the document. I hope this may provide a clearer understanding of the concepts.

Please let me know  if you think there are any details in the document that need clarification, we can continue to discuss improvements.

Best regards